### ISSUE TITLE
Staging Site Synchronization: If backup or restore job fails Calypso state is stuck in Synchronization

### ISSUE THREAD

**Issue by ash1eygrace** [Automattic/wp-calypso#83614]
### Quick summary

If a Jetpack backup or restore job fails during a staging site synchronization, Calypso doesn't get the memo that the restore failed and it's stuck in the synchronization state. 

### Steps to reproduce

1. Create a WordPress.com site with a Business or eCommerce plan. 
2. [Activate hosting access](https://wordpress.com/support/hosting-configuration/#activate-hosting-access)
3. [Create a staging site ](https://wordpress.com/support/how-to-create-a-staging-site/)
4. Make any arbitrary changes on the staging site, e.g. [change themes](https://wordpress.com/support/themes/choose-a-theme/). 
5. [Start a synchronization from the staging site to production](https://wordpress.com/support/how-to-create-a-staging-site/#synchronize-staging-to-production)
6. Make the backup or restore fail (e.g. break the jetpack connection manually or cause a critical error, or kill the Jetpack backup or restore job using internal tools etc.) 

### What you expected to happen

I expect Calypso to tell me that synchronization failed and to try again or contact support if the problem continues to occur. 

### What actually happened

The staging site Hosting Configuration > Calypso screen is stuck in a synchronization state: 

![image](https://github.com/Automattic/wp-calypso/assets/29527450/2eccbc66-1455-474a-be70-34f3e80464b4)

And the source site Hosting Configuration > Calypso screen is stuck in a synchronization state: 

![image](https://github.com/Automattic/wp-calypso/assets/29527450/cfdcee57-a2cb-4b07-9e33-bbc7ae71f5c1)

### Impact

All

### Available workarounds?

Yes, difficult to implement

No user-facing workaround. 
Steps to fix using internal tools are here: p9F6qB-dDE-p2

### Platform (Simple and/or Atomic)

Atomic

### Logs or notes

- 7229625-zd-a8c
- p9F6qB-dDE-p2

**Comment by rinazrina** [comment]
Another instance: 7233911-zd-a8c

Unable to delete the staging site as the synchronization seems to be stuck. 

Site reset didn't work. I had to revert the site and remove the subscription from SA to be able to delete the staging site.

**Comment by wojtekn** [comment]
@ash1eygrace @rinazrina we are fixing it under https://github.com/Automattic/wp-calypso/pull/83587

**Comment by cuemarie** [comment]
📌 REPRODUCTION RESULTS
- Tested on Atomic – **Replicated**

📌 FINDINGS/SCREENSHOTS/VIDEO
Replicated as described. I broke the connection by deactivating Jetpack on the staging site using internal tools.  Calypso continued to show the progress bar moving on the staging and production sites, eventually getting stuck mid-way. 
- I then reconnected JP on the staging site via JPDB, and calypso cleared out the sync status bars on both sites.

---

From here, I tried starting a new sync, and was shown a message to contact support instead:

![Markup on 2023-10-30 at 12:50:04](https://github.com/Automattic/wp-calypso/assets/27249804/d3bdc375-d97f-4b04-948d-f79d8af5d31e)

This may reflect the work happening in https://github.com/Automattic/wp-calypso/pull/83587

📌 ACTIONS
- Triaged
